hasura/metadata/databases/default/tables/public_membership.yaml

```diff
@@ -56,3 +57,11 @@ delete_permissions:
   - role: service
     permission:
       filter: {}
+  - role: user
+    permission:
+      filter:
+        _and:
+          - team_id:
+              _eq: X-Hasura-Team-Id
+          - role:
+              _eq: OWNER
```

Comment:
This means rows where team id matches and has a member that is an owner? Need the user-id here in an and with the role?

Reply:
Any OWNER in this team, as I can see it.

Reply:
But does this check if the user submitting the delete request is the Owner? 
Think it's missing a 
```
- user_id:
     _eq: X-Hasura-User-Id
```

Reply:
For some reason I keep reading this as it filters for rows where you are on the team and the role is owner. So instead of it saying you are on the team and the owner so you can delete any role on the team.
It's saying you are on the team and the role you are deleting is an owner role

Reply:
I think you're right @andy-t-wang, this seems to be looking at the role in the row, not the role of the user

Reply:
> It's saying you are on the team and the role you are deleting is an owner role
Just checked, seems you are right. But the problem is that if we will add check for `X-Hasura-User-Id` it will allow remove only the rows with you as member.

In this case I think we need to leave only 
```
- team_id
  _eq: X-Hasura-Team-Id
 ```

And then just show Remove button only for the ADMIN and OWNER roles from the code. This is already implemented in #457.

Reply:
@pdtfh @andy-t-wang Please check [this](https://github.com/worldcoin/developer-portal/pull/456/commits/a61ac17852622c5b5cd639cd8a69483e1de49589) commit

Reply:
I don't think this is the best approach, permissions as they currently are would allow any teammate to delete a membership record. for update permissions, I believe we're still letting any team member update any record that belongs to an OWNER, regardless of the permission level of the user executing.

In addition, do we plan to validate the `x-hasura-team-id` in the `api/v1/graphql` endpoint ? because in other models we rely on permissions to validate this, but here you could have an incorrect value and bypass permissions

Reply:
> I don't think this is the best approach, permissions as they currently are would allow any teammate to delete a membership record. 
How MEMBER can do this if he has nothing to click to invoke deletion? 

As additional measures I can add checks for roles: `{_or: [{role: {_eq: MEMBER}, {role: {_eq: ADMIN}}]}`. This way there is no possibility to remove OWNER.

> In addition, do we plan to validate the x-hasura-team-id in the api/v1/graphql endpoint ?
Can you clarify?